build/gen_stub.php

```diff
@@ -1262,29 +1264,89 @@ function initPhpParser() {
 }
 
 $optind = null;
-$options = getopt("fh", ["force-regeneration", "parameter-stats", "help"], $optind);
+$options = getopt("fh", ["force-regeneration", "parameter-stats", "help", "verify"], $optind);
 
 $context = new Context;
 $printParameterStats = isset($options["parameter-stats"]);
-$context->forceRegeneration =
-    isset($options["f"]) || isset($options["force-regeneration"]) || $printParameterStats;
+$verify = isset($options["verify"]);
+$context->forceRegeneration = isset($options["f"]) || isset($options["force-regeneration"]);
+$context->forceParse = $context->forceRegeneration || $printParameterStats || $verify;
 
 if (isset($options["h"]) || isset($options["help"])) {
-    die("\nusage: gen-stub.php [ -f | --force-regeneration ] [ --parameter-stats ] [ -h | --help ] [ name.stub.php | directory ]\n\n");
+    die("\nusage: gen-stub.php [ -f | --force-regeneration ] [ --parameter-stats ] [ --verify ] [ -h | --help ] [ name.stub.php | directory ]\n\n");
 }
 
+$fileInfos = [];
 $location = $argv[$optind] ?? ".";
 if (is_file($location)) {
     // Generate single file.
-    processStubFile($location, $context);
+    $fileInfo = processStubFile($location, $context);
+    if ($fileInfo) {
+        $fileInfos[] = $fileInfo;
+    }
 } else if (is_dir($location)) {
-    processDirectory($location, $context);
+    $fileInfos = processDirectory($location, $context);
 } else {
     echo "$location is neither a file nor a directory.\n";
     exit(1);
 }
 
 if ($printParameterStats) {
-    arsort($context->parameterStats);
-    echo json_encode($context->parameterStats, JSON_PRETTY_PRINT), "\n";
+    $parameterStats = [];
+
+    foreach ($fileInfos as $fileInfo) {
+        foreach ($fileInfo->getAllFuncInfos() as $funcInfo) {
+            foreach ($funcInfo->args as $argInfo) {
+                if (!isset($context->parameterStats[$argInfo->name])) {
+                    $parameterStats[$argInfo->name] = 0;
+                }
+                $parameterStats[$argInfo->name]++;
+            }
+        }
+    }
+
+    arsort($parameterStats);
+    echo json_encode($parameterStats, JSON_PRETTY_PRINT), "\n";
+}
+
+if ($verify) {
+    $errors = [];
+    $funcMap = [];
+    $aliases = [];
+
+    foreach ($fileInfos as $fileInfo) {
+        foreach ($fileInfo->getAllFuncInfos() as $funcInfo) {
+            /** @var FuncInfo $funcInfo */
+            $funcMap[$funcInfo->name->__toString()] = $funcInfo;
+
+            if ($funcInfo->aliasType === "alias") {
+                $aliases[] = $funcInfo;
+            }
+        }
+    }
+
+    foreach ($aliases as $alias) {
+        if ($alias->alias === null || !isset($funcMap[$alias->alias->__toString()])) {
```

Comment:
Can `alias` be null if aliasType is set?

Reply:
No, it can't. I just added the first condition to suppress PHPStorm for complaining about accessing the nullable `$alias->alias`. I can remove it if you find it useless.

Reply:
Yes, I think so. However, I noticed in the meanwhile that it doesn't complain anymore. ^^ Probably I messed something else up before. :)